src/main/java/org/everit/osgi/dev/maven/statistic/GoogleAnalyticsTrackingServiceImpl.java

```diff
@@ -0,0 +1,170 @@
+     * @param executedGoalName
+     *          the name of the executed goal.
+     * @param macAddressHash
+     *          the MAC address hash to anonym_user_id dimension.
+     */
+    EventSender(final String analyticsReferer, final String executedGoalName,
+        final String macAddressHash) {
+      this.analyticsReferer = analyticsReferer;
+      this.executedGoalName = executedGoalName;
+      this.macAddressHash = macAddressHash;
+      httpClient = new DefaultHttpClient();
+    }
+
+    @Override
+    public void run() {
+      HttpPost post = new HttpPost(GA_ENDPOINT);
+      List<NameValuePair> params = new ArrayList<>();
+      params.add(new BasicNameValuePair("v", "1")); // version
+      params.add(new BasicNameValuePair("tid", trackingId)); // tracking id
+      params.add(new BasicNameValuePair("cid", macAddressHash)); // client id
+      params.add(new BasicNameValuePair("t", "event")); // hit type
+      params.add(new BasicNameValuePair("ec", analyticsReferer)); // event category
+      params.add(new BasicNameValuePair("ea", executedGoalName)); // event action
+      params.add(new BasicNameValuePair("dimension1", macAddressHash)); // anonym_user_id
+      try {
+        UrlEncodedFormEntity entity =
+            new UrlEncodedFormEntity(params, "UTF-8");
+        post.setEntity(entity);
+        httpClient.execute(post);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  private static final String GA_ENDPOINT =
+      "http://www.google-analytics.com/collect?payload_data";
+
+  private static final String UNKNOWN_MAC_ADDRESS = "UNKNOWN_MAC_ADDRESS";
+
+  private final long analyticsWaitingTimeInMs;
+
+  private final ConcurrentMap<Long, Thread> sendingEvents = new ConcurrentHashMap<>();
+
+  private final boolean skipTracking;
+
+  private final String trackingId = "UA-69304815-1";
+
+  public GoogleAnalyticsTrackingServiceImpl(final long analyticsWaitingTimeInMs,
+      final boolean disableTracking) {
+    this.analyticsWaitingTimeInMs = analyticsWaitingTimeInMs;
+    skipTracking = disableTracking;
+  }
+
+  @Override
+  public void cancelEventSending(final long eventId) {
+    if (skipTracking) {
+      return;
+    }
+
+    Thread thread = sendingEvents.get(eventId);
+    if ((thread != null) && thread.isAlive()) {
+
+      try {
+        thread.join(analyticsWaitingTimeInMs);
+        if (thread.isAlive()) {
+          thread.interrupt();
+        }
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+
+    }
+  }
+
+  private String getMacAddressHash() {
+    try {
+      Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+      if (networkInterfaces.hasMoreElements()) {
+        NetworkInterface network = networkInterfaces.nextElement();
+        byte[] macAddress = network.getHardwareAddress();
+        byte[] encodedMacAddress = Base64.encodeBase64(macAddress);
```

Comment:
Base64 is not a hash algorithm, it can be decoded. We do not want to know the mac addresses of the plugin users, we just want to identify them. Use an SHA algorithm instead.